Client `query_timeout` leads to 400 invalid request error

After upgrading to fauna-python v0.7.0, I am now getting the following error when setting a timeout in the client and making requests:

```
E                   fauna.errors.errors.InvalidRequestError: 400: invalid_request
E                   Invalid header 'x-query-timeout-ms': Invalid number '5000.0'.
```

The client instantiation code is:

```python
def create_fauna_client(secret):
    """Create the FQLv10 client and return it.

    This should be called handler, or subsequently invoked functions within the handler,
    so when the lambda context is frozen, the socket connections are not lost.

    See: https://docs.fauna.com/fauna/current/drivers/known_issues
    """
    logger.debug("Initializing FaunaDB client.")
    # This timeout is currently set to half of the configured lambda timeout value. If
    # the terraform code is updated, this should be revisited.
    client = Client(secret=secret, query_timeout=timedelta(seconds=5))
    logger.debug("FaunaDB client initialized.")
    return client
```

--- a/.github/workflows/yapf.yml
+++ b/.github/workflows/yapf.yml
@@ -22,7 +22,7 @@ jobs:
           cache: pip
 
       - name: Install yapf
-        run: pip install yapf
+        run: pip install yapf==0.40.1
 
       - name: Check Format
         run: |

--- a/README.rst
+++ b/README.rst
@@ -172,7 +172,7 @@ If you want to run fauna, then run integration tests separately:
 
 .. code-block:: bash
 
-    $ make docker-fauna
+    $ make run-fauna
     $ source venv/bin/activate
     $ make install
     $ make integration-test

--- a/fauna/__init__.py
+++ b/fauna/__init__.py
@@ -1,5 +1,5 @@
 __title__ = "Fauna"
-__version__ = "0.7.0"
+__version__ = "0.7.1"
 __api_version__ = "10"
 __author__ = "Fauna, Inc"
 __license__ = "MPL 2.0"

--- a/fauna/client/client.py
+++ b/fauna/client/client.py
@@ -86,7 +86,7 @@ class Client:
             self._query_tags.update(query_tags)
 
         if query_timeout is not None:
-            self._query_timeout_ms = query_timeout.total_seconds() * 1000
+            self._query_timeout_ms = int(query_timeout.total_seconds() * 1000)
         else:
             self._query_timeout_ms = None
 

--- a/setup.py
+++ b/setup.py
@@ -19,7 +19,7 @@ requires = [
 ]
 
 extras_require = {
-    "lint": ["yapf==0.32.0"],
+    "lint": ["yapf==0.40.1"],
     "test": [
         "pytest==7.3.0", "pytest-env==0.8.1", "pytest-cov==4.0.0",
         "pytest-httpx==0.21.3", "pytest-subtests==0.10.0"

--- a/tests/integration/test_client.py
+++ b/tests/integration/test_client.py
@@ -1,4 +1,6 @@
+from datetime import timedelta
 import pytest
+from fauna.client.client import QueryOptions
 
 from fauna.errors import ClientError
 
@@ -27,3 +29,11 @@ def test_handle_invalid_json_response():
     with pytest.raises(ClientError, match=err_msg):
         c = Client(endpoint="https://dashboard.fauna.com/")
         c.query(fql("'foolery'"))
+
+
+def test_fauna_accepts_query_timeout_header():
+    c1 = Client(query_timeout=timedelta(seconds=5))
+    c1.query(fql('"hello"'))
+
+    c2 = Client()
+    c2.query(fql('"hello"'), QueryOptions(query_timeout=timedelta(seconds=5)))

--- a/tests/integration/test_query.py
+++ b/tests/integration/test_query.py
@@ -27,23 +27,28 @@ def test_query_smoke_test(subtests, client):
 
 def test_query_with_all_stats(client, a_collection):
     res = client.query(fql("${col}.create({})", col=a_collection))
+    doc_id = res.data.id
+    res = client.query(
+        fql("${col}.create({})\n${col}.byId(${id})",
+            col=a_collection,
+            id=doc_id))
     assert res.stats.compute_ops > 0
     assert res.stats.read_ops > 0
     assert res.stats.write_ops > 0
     assert res.stats.storage_bytes_read > 0
     assert res.stats.storage_bytes_write > 0
     assert res.stats.query_time_ms > 0
-    assert res.stats.contention_retries >= 0
+    assert res.stats.contention_retries == 0
 
 
 def test_query_with_constraint_failure(client):
     with pytest.raises(QueryRuntimeError) as e:
         client.query(
-            fql('Function.create({"name": "double", "body": "x => x * 2"})'))
+            fql('Function.create({"name": "false", "body": "_ => false"})'))
 
     assert e.value.constraint_failures == [
         ConstraintFailure(
-            message="The identifier `double` is reserved.",
+            message="The identifier `false` is reserved.",
             name=None,
             paths=[["name"]],
         ),

